src/server.py

```diff
@@ -0,0 +1,88 @@
+"""Server for Echo server assignment."""
+
+from __future__ import print_function
+import socket
+
+
+def server():
+    serv = socket.socket(socket.AF_INET, socket.SOCK_STREAM, socket.IPPROTO_TCP)
+    port = 5001
+    address = ('127.0.0.1', port)
+    serv.bind(address)
+
+    serv.listen(1)
+    print("Listening on: ", port)
+
+    while True:
+        conn, addr = serv.accept()
+        try:
+            req_string = b''
+            buffer_length = 10
+            while req_string[-8:] != b"\\r\\n\\r\\n":
+                part = conn.recv(buffer_length)
+                req_string += part
+                print("Received: ", part)
+
+            print("Testing, ", req_string)
+            try:
+                req_result = parse_request(req_string)
+                req_result += b'\\r\\n\\r\\n'
+                conn.sendall(response_ok())
+                conn.sendall(req_result)
+            except TypeError:
+                print("Sending Error response.")
+                conn.sendall(response_err())
+
+            print('waiting')
+            conn.close()
+
+        except KeyboardInterrupt:
+            print("Shutting down server.")
+            break
+    conn.close()
```

Comment:
do these two statements make sense here?  When should the connection close, and when should the server itself shutdown?